go-controller/pkg/clustermanager/egressip_controller.go

```diff
@@ -1042,6 +1042,58 @@ func (eIPC *egressIPClusterController) reconcileEgressIP(old, new *egressipv1.Eg
 	return nil
 }
 
+func (eIPC *egressIPClusterController) syncEgressIPs(ips []interface{}) error {
+	if !util.PlatformTypeIsEgressIPCloudProvider() {
+		return nil
+	}
+	cloudPrivateIPConfigMap, err := eIPC.getCloudPrivateIPConfigMap()
+	if err != nil {
+		return fmt.Errorf("syncEgressIPs unable to get cloud private ip config: %w", err)
+	}
+	egressIPs, err := eIPC.kube.GetEgressIPs()
+	if err != nil {
+		return fmt.Errorf("syncEgressIPs unable to get Egress IPs: %w", err)
+	}
+	for _, egressIP := range egressIPs.Items {
+		updatedStatus := []egressipv1.EgressIPStatusItem{}
+		ipNotFound := false
+		for _, status := range egressIP.Status.Items {
+			cloudPrivateIPConfigName := ipStringToCloudPrivateIPConfigName(status.EgressIP)
+			if nodeName, exists := cloudPrivateIPConfigMap[cloudPrivateIPConfigName]; exists && status.Node == nodeName {
+				updatedStatus = append(updatedStatus, status)
+			} else {
+				// Set ipNotFound flag to true because egress ip entry not found or not set with correct node name in
+				// cloud private ip config object.
+				ipNotFound = true
+			}
+		}
+		if ipNotFound {
+			// There could be one or more stale entry found in egress ip object, remove it by patching egressip
+			// object with updated status.
+			err = eIPC.patchReplaceEgressIPStatus(egressIP.Name, updatedStatus)
```

Comment:
the watchers for egressIPs have not yet started at this stage right? so when we start getting the events for adds, its going to be this updated list right?

Reply:
the reason why we had the logic called from namespaces is because that is the first watcher we start, so if we had put it in egressIP watcher it would be too late.. but here I think the only 2 objects you care about are egressIPs and CloudPrivateIPs right? Since you don't care about namespaces and pods in this instance, whichever Watcher is started first in CM for EIP we should call it from there to be safe.